runtime/wasm/src/module.rs

```diff
@@ -313,8 +313,10 @@ where
         let unresolved_call: UnresolvedContractCall = self.heap.asc_get(call_ptr);
 
         info!(self.logger, "Call smart contract";
+              "function" => &unresolved_call.function_name,
+              "address" => &unresolved_call.block_hash.to_string(),
```

Comment:
This isn't correct. You're passing in the block hash but call it the contract address? Perhaps you want both?